lib/src/erc20/erc20_base_subsystem.cairo

```diff
@@ -0,0 +1,131 @@
+#[contract]
+mod ERC20BaseSubsystem {
+
+    use dojo::world;
+    use zeroable::Zeroable;
+    use starknet::get_caller_address;
+    use starknet::contract_address_const;
+    use starknet::ContractAddress;
+    use starknet::ContractAddressZeroable;
+
+    struct Storage {
+        name: felt252,
+        symbol: felt252,
+        decimals: u8,
+        total_supply: u256,
+        balances: OwnershipComponent,
+        allowances: ApprovalComponent,
```

Comment:
```suggestion
```

Component state is stored in the world

Reply:
I'm not sure of how to put the "contract" erc20 and Components working together in the world, would like to see it with you